cmd/objectstore/server.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"github.com/filedag-project/filedag-storage/http/objectstore/iamapi"
+	"github.com/filedag-project/filedag-storage/http/objectstore/s3api"
+	"github.com/filedag-project/filedag-storage/http/objectstore/uleveldb"
+	"github.com/filedag-project/filedag-storage/http/objectstore/utils"
+	"github.com/gorilla/mux"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/urfave/cli/v2"
+	"net/http"
+	"os"
+)
+
+var log = logging.Logger("sever")
+
+const (
+	deFaultDBFILE      = "/tmp/leveldb2/fds.db"
+	defaultPort        = ":9985"
+	fileDagStoragePort = "FILE_DAG_STORAGE_PORT"
+	dbPath             = "DBPATH"
+)
+
+//startServer Start a IamServer
+func startServer() {
+	uleveldb.DBClient = uleveldb.OpenDb(os.Getenv(dbPath))
+	router := mux.NewRouter()
+	iamapi.NewIamApiServer(router)
+	s3api.NewS3Server(router)
+
+	for _, ip := range utils.MustGetLocalIP4().ToSlice() {
+		log.Infof("start sever at http://%v%v", ip, os.Getenv(fileDagStoragePort))
+	}
+	err := http.ListenAndServe(os.Getenv(fileDagStoragePort), router)
+	if err != nil {
+		log.Errorf("Listen And Serve err%v", err)
+		return
+	}
+	defer uleveldb.DBClient.Close()
+}
```

Comment:
How does OpenDb handle error cases? The defer statement should be written after DB has been successfully opened.

Reply:
This was my oversight and will be fixed soon